src/build/env.py

```diff
@@ -169,20 +179,26 @@ def _create_isolated_env_venv(path):  # type: (str) -> Tuple[str, str]
     import venv
 
     venv.EnvBuilder(with_pip=True).create(path)
-    executable, script_dir = _find_executable_and_scripts(path)
-    # avoid the setuptools from ensurepip to break the isolation
-    if sys.version_info < (3, 6, 6):  # python 3.5 up to 3.6.5 come with pip 9 that's too old, for new standards
+    executable, script_dir, purelib = _find_executable_and_scripts(path)
+
+    # Require a minimum version of pip (currently 10)
+    (pip_version,) = (
```

Comment:
I don't think you need the left side parenthesis

Reply:
I believe Black requires this for readability, the comma is easy to miss. Though I've gotten used to writing it due to thinking Black requires it.

Reply:
Well also this is wasteful you exhaust the distributions irrespective of where pip is found (first or last). Use instead ``next`` to stop at the first, and then you don't need this tuple unrolling.

Reply:
These three lines read to me quite bad 😿 I'd prefer to have a one-liner instead.

Reply:
Was just hoping to get a little input from @FFY00 so I don't end up having to backtrack, but I guess I could backtrack if needed...

Reply:
The is one line now, though, isn't it?

```python
(pip_version,) = (Version(d.version) for d in metadata.distributions(name='pip', path=[purelib]))
```

Reply:
I guess I just don't like using tuple unrolling for one element. Looks odd 🤷‍♂️

Reply:
Much prefer next over a generator 👍

Reply:
Now that I have that version recorded above, I can change the one in the PR. ;)

Reply:
Do I really have to stick an `iter()` in here to make mypy happy? Technically, I guess that is correct, you should not iterate over something with `next()` without calling `iter()`.
```
src/build/env.py:184: error: No overload variant of "next" matches argument type "Iterable[Distribution]"
src/build/env.py:184: note: Possible overload variant:
src/build/env.py:184: note:     def [_T] next(Iterator[_T]) -> _T
src/build/env.py:184: note:     <1 more non-matching overload not shown>
```

Reply:
So it's:

```python
pip_version = Version(next(iter(metadata.distributions(name='pip', path=[purelib]))).version)
```

Saves about 4 characters, and has _way_ too many parentheses.